src/pybind/mgr/prometheus/module.py

```diff
@@ -385,13 +395,24 @@ def get_osd_stats(self):
                 self.metrics.append('osd_{}'.format(stat), val,
                                     ('osd.{}'.format(id_),))
 
+    def get_service_list(self):
+        ret = {}
+        for server in self.list_servers():
+            version = server.get('ceph_version', '')
+            host = server.get('hostname', '')
+            for service in server.get('services', []):
+                ret.update({(service['id'], service['type']): (host, version)})
```

Comment:
Why don't you just assign the value with `ret[(service['id'], service['type'])] = ` directly?

Reply:
I find using update, the code is a bit more readable, i.e. the structure of ```ret``` is more obvious. But its a matter of taste.